front/components/assistant/conversation/input_bar/editor/MentionWithPaste.tsx

```diff
@@ -17,7 +17,8 @@ export const MentionWithPaste = Mention.extend({
           (suggestion) => {
             return [
               ...text.matchAll(
-                new RegExp(escapeRegExp("@" + suggestion.label), "g")
+                // Note: matching the @ that are found either at the start of a line or after a whitespace character.
+                new RegExp("(^@|\\s@)" + escapeRegExp(suggestion.label), "g")
```

Comment:
double backslash ?

Reply:
Since we are using the `new Regexp` syntax instead of the `//` one I believe we have to escape the / in the string for the Regexp to then process it as a `\s`.